src/server/helpers/achievement.js

```diff
@@ -474,71 +454,72 @@ function achievementToUnlockId(achievementUnlock) {
  * @param {int} revisionId - Revision to get release date of
  * @returns {int} - Days since edition was released
  */
-function getEditionDateDifference(orm, revisionId) {
+async function getEditionDateDifference(orm, revisionId) {
 	const {EditionRevision} = orm;
-	return new EditionRevision({id: revisionId}).fetch()
-		.then((edition) => edition.related('data').fetch())
-		.then((data) => data.related('releaseEventSet').fetch())
-		.then(
-			(releaseEventSet) =>
-				releaseEventSet.related('releaseEvents').fetch()
-		)
-		.then((releaseEvents) => {
-			let differencePromise;
-			if (releaseEvents.length > 0) {
-				const msInOneDay = 86400000;
-				const attribs = releaseEvents.models[0].attributes;
-				const date = new Date(
-					attribs.year,
-					attribs.month - 1,
-					attribs.day
-				);
-				const now = new Date(Date.now());
-				differencePromise =
-					Math.round((date.getTime() - now.getTime()) / msInOneDay);
-			}
-			else {
-				differencePromise =
-				new Promise((resolve, reject) => reject(new Error('no date attribute')));
-			}
-			return differencePromise;
-		})
-		.catch(() => new Promise((resolve, reject) => reject(new Error('no date attribute'))));
+	try {
+		const edition = await new EditionRevision({id: revisionId}).fetch();
+		const data = await edition.related('data').fetch();
+		const releaseEventSet = await data.related('releaseEventSet').fetch();
+		const releaseEvents = await releaseEventSet.related('releaseEvents').fetch();
```

Comment:
I feel like we should be able to replace this with the right ORM relationships.
I wonder if 
```suggestion
		const edition = await new EditionRevision({id: revisionId}).fetch({withRelated: 'data.releaseEventSet.releaseEvents'});
		const releaseEvents = edition.related('releaseEventSet').related('releaseEvents');
```
Or something along those lines should work. I'm not sure about the syntax for the second line (I'm not 100% certain we can chain the calls like that). 

If all else fails you can use `edition.toJSON()` and then access `edition.releaseEventSet.releaseEvents`.
In that case there will be differences in the code below too; `releaseEvents.models[0].attributes`  will probably be something like `releaseEvents[0]`.

Reply:
> Or something along those lines should work. I'm not sure about the syntax for the second line (I'm not 100% certain we can chain the calls like that).

Chaining isn't working, though the second approach works perfect :)